File: etamen-backend/app/Modules/Labs/Infrastructure/Models/LabOrderItem.php
<?php

namespace App\Modules\Labs\Infrastructure\Models;

use App\Modules\Labs\Domain\Enums\LabOrderItemType;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class LabOrderItem extends Model
{
    protected $fillable = [
        'order_id',
        'item_type',
        'test_id',
        'package_id',
        'item_name',
        'unit_price',
        'quantity',
        'line_total',
    ];

    protected function casts(): array
    {
        return [
            'item_type' => LabOrderItemType::class,
            'unit_price' => 'decimal:2',
            'quantity' => 'integer',
            'line_total' => 'decimal:2',
        ];
    }

    public function order(): BelongsTo
    {
        return $this->belongsTo(LabOrder::class, 'order_id');
    }

    public function test(): BelongsTo
    {
        return $this->belongsTo(LabTest::class);
    }

    public function package(): BelongsTo
    {
        return $this->belongsTo(LabPackage::class);
    }
}

File: etamen-flutter/lib/features/labs/domain/entities/lab_order_item.dart
class LabOrderItem {
  const LabOrderItem({
    required this.itemType,
    required this.itemName,
    required this.quantity,
    required this.unitPrice,
    required this.lineTotal,
    this.testId,
    this.packageId,
  });

  final String itemType;
  final int? testId;
  final int? packageId;
  final String itemName;
  final int quantity;
  final String unitPrice;
  final String lineTotal;
}
